libeos-updater-util/avahi-service-file.h

```diff
@@ -44,4 +46,42 @@ gboolean eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
                                         GCancellable  *cancellable,
                                         GError       **error);
 
+/* new DNS-SD records format for ostree */
+
+#define EOS_OSTREE_AVAHI_SERVICE_TYPE "_repo._ostree._tcp"
+
+#define EOS_OSTREE_AVAHI_VERSION_FIELD "v"
+#define EOS_OSTREE_AVAHI_VERSION_VARIANT_TYPE G_VARIANT_TYPE_BYTE
+
+#define EOS_OSTREE_AVAHI_V1_REFS_BLOOM_FILTER_FIELD "rb"
+#define EOS_OSTREE_AVAHI_V1_REFS_BLOOM_FILTER_VARIANT_TYPE (G_VARIANT_TYPE ("(yyay)"))
+#define EOS_OSTREE_AVAHI_V1_SUMMARY_TIMESTAMP_FIELD "t"
+#define EOS_OSTREE_AVAHI_V1_SUMMARY_TIMESTAMP_VARIANT_TYPE G_VARIANT_TYPE_UINT64
+#define EOS_OSTREE_AVAHI_V1_REPOSITORY_INDEX_FIELD "ri"
+#define EOS_OSTREE_AVAHI_V1_REPOSITORY_INDEX_VARIANT_TYPE G_VARIANT_TYPE_UINT16
+
+#define EOS_OSTREE_AVAHI_OPTION_FORCE_VERSION_Y "force-version"
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_HASH_ID_Y "bloom-hash-id"
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_K_Y "bloom-k"
+#define EOS_OSTREE_AVAHI_OPTION_BLOOM_SIZE_U "bloom-size"
+#define EOS_OSTREE_AVAHI_OPTION_REPO_INDEX_Q "repository-index"
+#define EOS_OSTREE_AVAHI_OPTION_PORT_Q "port"
+
+#define EOS_OSTREE_AVAHI_STR_BLOOM_HASH_ID 1
+
+gboolean eos_ostree_avahi_service_file_generate (const gchar   *avahi_service_directory,
+                                                 OstreeRepo    *repo,
+                                                 GVariant      *options,
+                                                 GCancellable  *cancellable,
+                                                 GError       **error);
+gboolean eos_ostree_avahi_service_file_delete (const gchar   *avahi_service_directory,
+                                               guint16        repository_index,
+                                               GCancellable  *cancellable,
+                                               GError       **error);
+gboolean eos_ostree_avahi_service_file_cleanup_directory (const gchar   *avahi_service_directory,
```

Comment:
Right, I also had doubts about this API. I will make it simpler.

Also, avahi currently uses inotify (or kqueue on BSDs I think) to watch the services directory. I don't see any timeout here, so I think avahi reloads the config immediately.